chargeio/services/Merchant/MerchantService.cs

```diff
@@ -1,16 +1,16 @@
-﻿using ChargeIo.Infrastructure;
-using ChargeIo.Models;
+﻿using ChargeIO.Infrastructure;
+using ChargeIO.Models;
 
-namespace ChargeIo.Services.Merchant
+namespace ChargeIO.Services.Merchant
 {
 	public class MerchantService
 	{
-        private string SecretKey { get; set; }
+        private string SecretKey { get; }
         
-        public MerchantService(string secretKey = null)
-		{
-            SecretKey = secretKey ?? Configuration.SecretKey;
-		}
+        public MerchantService(string secretKey = "")
+        {
+            SecretKey = secretKey.Length > 0 ? secretKey : Configuration.SecretKey;

```

Comment:
If the given secretKey argument is null, won't this now raise an exception? In which case, it just needs to be guarded to reject a null value.